File: gaia-sdk-javascript/src/graphql/response/type/SkillStatus.ts

import {Failure} from "./Failure";

import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../../request/enumeration/Order";
import {OrderByField} from "../../request/enumeration/OrderByField";
import {EdgeOrderByField} from "../../request/enumeration/EdgeOrderByField";

export interface SkillStatus {
    health?:string, 
    running?:number, 
    pending?:number, 
    failures?:[Failure]
}

File: gaia-sdk-java/gaia-sdk-graphql/src/main/kotlin/gaia/sdk/request/type/SkillStatus.kt
package gaia.sdk.request.type

import gaia.sdk.client.Type
import gaia.sdk.request.intf.*
import gaia.sdk.client.Input
import gaia.sdk.Uuid
import gaia.sdk.ISO8601
import gaia.sdk.Struct
import gaia.sdk.request.input.*
import gaia.sdk.request.enumeration.*

class SkillStatus: Type() {

    fun health() { 
        add {"health" } 
    }

    fun running() { 
        add {"running" } 
    }

    fun pending() { 
        add {"pending" } 
    }

    fun failures(config: Failure.() -> Unit) = 
        add { "failures{ " + Failure().apply(config).render(it) + "}"}

}

